modules/unsupported/csv/src/main/java/org/geotools/data/csv/parse/CSVSpecifiedLatLngStrategy.java

```diff
@@ -68,20 +78,76 @@ protected SimpleFeatureType buildFeatureType() {
                 csvReader.close();
             }
         }
-        SimpleFeatureTypeBuilder builder = CSVStrategySupport.createBuilder(csvFileState, headers,
+        SimpleFeatureTypeBuilder builder = CSVStrategy.createBuilder(csvFileState, headers,
                 typesFromData);
         Class<?> latClass = typesFromData.get(latField);
         Class<?> lngClass = typesFromData.get(lngField);
-        if (CSVStrategySupport.isNumeric(latClass) && CSVStrategySupport.isNumeric(lngClass)) {
-            builder.remove(latField);
+        if (CSVStrategy.isNumeric(latClass) && CSVStrategy.isNumeric(lngClass)) {
+        	List<String> csvHeaders = Arrays.asList(headers);
+        	int index = csvHeaders.indexOf(latField);
+        	AttributeTypeBuilder builder2 = new AttributeTypeBuilder();
+        	builder2.setCRS(DefaultGeographicCRS.WGS84);
+        	builder2.binding(Point.class);       	
+        	AttributeDescriptor descriptor = builder2.buildDescriptor(pointField);
+        	builder.add(index, descriptor);
+			
+        	builder.remove(latField);
             builder.remove(lngField);
-            builder.add(pointField, Point.class);
         }
         return builder.buildFeatureType();
     }
-
+    
+    @Override
+    public void createSchema(SimpleFeatureType featureType) throws IOException {
+        List<String> header = new ArrayList<String>();
+        
+        GeometryDescriptor geometryDescrptor = featureType.getGeometryDescriptor();
+        if (geometryDescrptor != null
+                && CRS.equalsIgnoreMetadata(DefaultGeographicCRS.WGS84,
+                        geometryDescrptor.getCoordinateReferenceSystem())
+                && geometryDescrptor.getType().getBinding().isAssignableFrom(Point.class)) {
+            header.add("LAT");
+            header.add("LON");
```

Comment:
I find it odd that we are specifically asking the constructor for a custom "lat" and "lon" name ... and then ignoring it when making the schema?

If I was a user I would feel ignored :)